odoo/tools/image.py

```diff
@@ -117,6 +118,14 @@ def image_quality(self, quality=0, output_format=''):
         :return: image
         :rtype: bytes or False
         """
+        dbname = odoo.tools.config['db_name']
+        registry = odoo.registry(dbname)
+        if registry.ready:
+            with registry.cursor() as cr:
+                uid = odoo.SUPERUSER_ID
+                env = odoo.api.Environment(cr, uid, {})
+                if env['ir.config_parameter'].sudo().get_param('base.no_image_quality_optimization'):
+                    return self.source
```

Comment:
This doesn't look correct at all.

This file looks like a tools library for images, almost independent from odoo.
You may have a image_quality function calling this on on a model, or just pass the parameter where this is called.

As @ryv-odoo noticed, this wont work in multi db. dbname can be a comma separted list or empty if used with a db filter.

This will be a lot of work to adapt all image_process call but this solution is not correct. 

You should check the runbot run, changing the parameter on the all database and checking the base database, or just trying to make this work on the base database

Depending on your use case, an hackish solution would be to use request to get en environment if there is one.

Reply:
+1, you cannot import odoo inside of tools, using `odoo.http.request` is not a solution either. The function signature should be modified so add a parameter, the callers (e.g. ir.binary) then can set that parameter according to the ICP.